### PR TITLE
find_cuda: fix crash caused by unicode chars in ldconfig output

### PR DESCRIPTION
**Pull request by domschl**
`find_cuda_config.py` crashes, if Unicode chars are found in `ldconfig` output:
```
Traceback (most recent call last):
  File "third_party/gpus/find_cuda_config.py", line 463, in <module>
    main()
  File "third_party/gpus/find_cuda_config.py", line 455, in main
    for key, value in sorted(find_cuda_config().items()):
  File "third_party/gpus/find_cuda_config.py", line 418, in find_cuda_config
    _get_default_cuda_paths(cuda_version))
  File "third_party/gpus/find_cuda_config.py", line 159, in _get_default_cuda_paths
    ] + _get_ld_config_paths()
  File "third_party/gpus/find_cuda_config.py", line 139, in _get_ld_config_paths
    match = pattern.match(line.decode("ascii"))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 27: ordinal not in range(128)
Asking for detailed CUDA configuration...
```
Localized versions of `ldconfig` versions use Unicode chars, e.g.: `»/etc/ld.so.cache«`, causing the crash.

The fix simply skips lines with Unicode chars.

### CODE REVIEW

**Comment by jeythekey** [comment]
This line also fixed it for me, don't know if it is a (more?) generic one:
`match = pattern.match(line.decode(sys.stdin.encoding))`

**Comment by gbaned** [comment]
@skye can you please review this? Thanks!

File: third_party/gpus/find_cuda_config.py [hunk 1]
```diff
@@ -136,7 +136,10 @@ def _get_ld_config_paths():
   pattern = re.compile(".* => (.*)")
   result = set()
   for line in output.splitlines():
-    match = pattern.match(line.decode("ascii"))
+    try:
+      match = pattern.match(line.decode("ascii"))
+    except:
```
Comment (skye) [inline third_party/gpus/find_cuda_config.py]: except UnicodeDecodeError: ?

**Review by skye — DISMISSED** [review]
LGTM pending the one comment.

File: third_party/gpus/find_cuda_config.py (hunk 1 above)
Comment (domschl) [reply third_party/gpus/find_cuda_config.py]: Updated with UnicodeDecodeError.

**Review by skye — APPROVED** [review]

**Pull request merged**